plugins/notifications-backend-module-slack/api-report.md

```diff
@@ -0,0 +1,15 @@
+## API Report File for "@backstage/plugin-notifications-backend-module-slack"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+import { BackendFeature } from '@backstage/backend-plugin-api';
+
+// @public (undocumented)
+export const ANNOTATION_NOTIFICATION_SLACK_CHANNEL_ID =
+  'slack.com/notification-channel-id';
```

Comment:
yeah we can shorten the annotation name. Currently this is implemented using Slack Workflows with a Webhook Trigger. The channel ID is passed to the workflow which is specific to a workspace.

Since we'd have to parse out the workspace name as an additional routing param we might also want to shorten that:

e.g. `slack.com/notifications: my-workspace-name/C123456`

We could then augment the config schema to include workspace name for the processor instance.

Reply:
Just thinking a little down the road, but I can potentially foresee it being useful to send certain notification types to different channels. For example a team/group might want to have dedicated slack channels for different purposes. Internally we do this quite often.

Not exactly sure how this would translate to an annotation though as it would need to be at a minimum a list of key/values. Wondering if later on we could append the `spec` to include some sort of notification preferences. Feels a bit wrong, but not sure where else this would naturally fit.

Reply:
Yeah this is something else I've been thinking about. Generally we have private/public/alerts channels, and i think generally we would just use a single channel to route to, routing to multiple channels based on notification type could get complicated quickly.